xmtp_mls/src/api_client_wrapper.rs

```diff
@@ -34,6 +40,51 @@ where
         Self { api_client }
     }
 
+    pub async fn read_topic(
+        &self,
+        topic: &str,
+        start_time_ns: u64,
+    ) -> Result<Vec<Envelope>, ApiError> {
+        let mut cursor: Option<Cursor> = None;
+        let mut out: Vec<Envelope> = vec![];
+        let page_size = 100;
+        loop {
+            let result = self
+                .api_client
+                .query(QueryRequest {
+                    content_topics: vec![topic.to_string()],
+                    start_time_ns,
+                    end_time_ns: 0,
+                    paging_info: Some(PagingInfo {
+                        cursor,
+                        limit: page_size,
+                        direction: SortDirection::Ascending as i32,
+                    }),
+                })
+                .await?;
+
+            for envelope in &result.envelopes {
+                out.push(envelope.clone());
+            }
```

Comment:
Small nit: I think you can do `out.append(result.envelopes)` if you do `let mut result` above and store the count beforehand, that might save a number of clones

Reply:
That's pretty slick. Done